static/js/util/profile_history_edit.js

```diff
@@ -79,5 +79,5 @@ export function deleteWorkHistoryEntry () {
   const { saveProfile, profile, ui } = this.props;
   let clone = _.cloneDeep(profile);
   clone['work_history'].splice(ui.deletionIndex, 1);
-  saveProfile(employmentValidation, clone, ui);
+  return saveProfile(employmentValidation, clone, ui);
```

Comment:
ah ok I see, these functions were changed to return the `saveProfile` promise. looks good!